src/HasteModuleLoader/HasteModuleLoader.js

```diff
@@ -226,9 +226,12 @@ Loader.prototype._execModule = function(moduleObj) {
     || (onlyCollectFrom && onlyCollectFrom[modulePath] === true);
 
   if (shouldCollectCoverage) {
+    if (!this._CoverageCollector) {
+      this._CoverageCollector = require(this._config.coverageCollector);
+    }
```

Comment:
Hmm, that's odd.. The loader is new-ed up here:
https://github.com/facebook/jest/blob/master/src/TestRunner.js#L102
Which should have a non-test-environment context...

Maybe you're seeing this behavior in a test and just need to mock/fake/etc `require` in the test surrounding where this code is executed?